services/dynamic-sidecar/src/simcore_service_dynamic_sidecar/core/validation.py

```diff
@@ -125,27 +125,32 @@ def _gen_parts_env_vars(
     return [f"{k}={v}" for k, v in dict_spec_env_vars.items()]
 
 
-def _inject_backend_networking(
-    parsed_compose_spec: dict[str, Any], network_name: str = "__backend__"
-) -> None:
+DEFAULT_BACKEND_NETWORK_NAME = "__backend__"
+
+
+def _inject_backend_networking(parsed_compose_spec: dict[str, Any]) -> None:
     """
     Put all containers in the compose spec in the same network.
     The `network_name` must only be unique inside the user defined spec;
     docker-compose will add some prefix to it.
     """
 
-    networks = parsed_compose_spec.get("networks", {})
-    networks[network_name] = None
+    networks = parsed_compose_spec.setdefault("networks", {})
```

Comment:
i find very difficult to trace what is the logic here with defaults ... I guess you adapted the logic to the test cases ... but i wonder if there is a more clear way to write it

Reply:
Would this work?

```python

networks = parsed_compose_spec.setdefault("networks", {})
networks.setdefault(DEFAULT_BACKEND_NETWORK_NAME, {})
                   
```

Reply:
ok tried to make it more explicit